### ISSUE TITLE
How to export model or use checkpoint for separation

### ISSUE THREAD

**Issue by lyndonlauder** [KinWaiCheuk/demucs_lightning#11]
Hi sorry for my issues, I have trained a model and have a e=21-TRAIN_loss=0.10.ckpt file. How can i convert this file to a model or how do i use the checkpoint for separation on my own files?

Thanks for your help !

**Comment by heungky** [comment]
Hi, thank you for using demucs_lightning.

Given that you already have a checkpoint file which has trained model weight inside. You can use `test.py` and specify `resume_checkpoint` in CLI to evaluate the trained model performance. You can refer to [README.md](https://github.com/KinWaiCheuk/demucs_lightning#testing-pretrained-model) for more information.

Currently, we don't support custom dataset. Feel free to write your test set class for your own dataset and replace it [here](https://github.com/KinWaiCheuk/demucs_lightning/blob/master/test.py#L38). Keep in mind that the returned tensor dimension of your custom dataset class need to match with `(Batch_number, No. of channel, length of audio)`. You can refer to [AudioLoader_MusdbHQ](https://github.com/KinWaiCheuk/AudioLoader/blob/master/AudioLoader/music/mss/mss_README.md#musdbhq)

Hope it helps :)

**Comment by lyndonlauder** [comment]
@heungky thank you for your reply. In the original demucs repo you can export the model from the checkpoint file with this link https://github.com/facebookresearch/demucs/blob/main/docs/training.md#model-export

Is it possible to do this with your lightning repo? I would like to do this as I would like to release my model. Thank you for your reply :)

**Comment by KinWaiCheuk** [comment]
> @heungky @KinWaiCheuk I trained a model with save_weights_only=true but when i try to use the model to perform a separation with demucs I get this error
> 
> `FATAL: last is neither a single pre-trained model or a bag of models.`

Hi @lyndonlauder, are you trying to use the checkpoint generated from our repo and use it in the original demucs repo? We haven't tested this functionality yet, so it is very likely to be some compatibility issues between our version of demucs and the original version of demucs.

@heungky has just updated the code for inferencing, if you follow this [section ](https://github.com/KinWaiCheuk/demucs_lightning#inferencing)of README.md, you would be able to use your pretrained model obtained from our repo and perform source separation on your own audio files.

**Comment by lyndonlauder** [comment]
@KinWaiCheuk @heungky Hello, today i tested inferencing and i had two problem. I train my model with two sources (vocal, other).

When i do inferencing with a folder with just one track inside it only outputs one source (other.wav) instead of two.

Then when i do inferencing with a folder with multiple tracks it only separates the first track but multiple times, for example if i have three tracks inside my folder, it will just separate the first track three times instead of doing each track.

Sorry for my problems.

**Comment by KinWaiCheuk** [comment]
May I double check that you train your new two-source model by changing the configuration from `sources: ['drums', 'bass', 'other', 'vocals']` to `sources: ['other', 'vocals']`?

We will check what is wrong and get back to you as soon as possible.

Thanks for using our repo and reporting problems. It helps us to improve our repo further :)

**Comment by lyndonlauder** [comment]
@KinWaiCheuk Yes I changed sources: `['drums', 'bass', 'other', 'vocals']` to sources: `['vocals', 'other']` also i changed `weights: [1., 1., 1., 1.]` to `weights: [1., 1.]` I also changed the sources in [Audioloader](https://github.com/KinWaiCheuk/AudioLoader/blob/1851acadca94c035fa659930a4311984d05f955a/AudioLoader/music/mss/musdbhq.py#L33) from `sources= ['drums', 'bass', 'other', 'vocals']` to `sources= ['vocals', 'other']`

Thank you for your help

**Comment by heungky** [comment]
Hi @[lyndonlauder](https://github.com/lyndonlauder)  thank you for your feedback. 
I have just updated the code for inferencing and Audioloader_MusdbHQ. 

Now you can train the model with specific number of sources by using `sources` argument in MusdbHQ(). You can refer [here](https://github.com/heungky/AudioLoader/blob/master/AudioLoader/music/mss/mss_README.md#MusdbHQ)

Feel free to do a git pull and use the inferencing code again, it should be fine now :)
